app/models/katello/repository.rb

```diff
@@ -103,6 +103,8 @@ class Repository < Katello::Model
     scope :non_archived, where('environment_id is not NULL')
     scope :archived, where('environment_id is NULL')
 
+    scope :completer_scope, lambda { |options| self.send(options[self.table_name]) }
```

Comment:
Was hoping @jlsherrill might chime into my diff, but he pointed out that this would only allow the use of named scopes and not `where()` via a lambda. Not sure if we want to account for that here and now or not.